Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -1541,15 +1536,66 @@ - (void)initializeCollectionViewLayout
         self.collectionView.collectionViewLayout.messageBubbleLeftRightMargin = 80.0f;
     }
 
-    // Bubbles
     self.collectionView.collectionViewLayout.bubbleSizeCalculator = [OWSMessagesBubblesSizeCalculator new];
-    JSQMessagesBubbleImageFactory *bubbleFactory = [[JSQMessagesBubbleImageFactory alloc] init];
-    self.incomingBubbleImageData =
-        [bubbleFactory incomingMessagesBubbleImageWithColor:[UIColor jsq_messageBubbleLightGrayColor]];
-    self.outgoingBubbleImageData = [bubbleFactory outgoingMessagesBubbleImageWithColor:[UIColor ows_materialBlueColor]];
-    self.currentlyOutgoingBubbleImageData =
-        [bubbleFactory outgoingMessagesBubbleImageWithColor:[UIColor ows_fadedBlueColor]];
-    self.outgoingMessageFailedImageData = [bubbleFactory outgoingMessagesBubbleImageWithColor:[UIColor grayColor]];
+}
+
++ (JSQMessagesBubbleImageFactory *)sharedBubbleImageFactory
+{
+
```

Comment:
Should we assert that we're on the main thread in these accessors?  I _think_ we only need them on the main thread while configuring views and these factories might not be thread-safe.